.github/workflows/update_submodule.yml

```diff
@@ -39,6 +39,9 @@ jobs:
     steps:
       - name: Checkout the repo
         uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
+        with:
+          submodules: true
+          fetch-tags: true
 
       - name: Update submodule
         run: git -C "ext/${REPO_NAME}" checkout "${TAG_NAME}"
```

Comment:
Is this still necessary? It kinda seems to me like the two things do the same thing (or at least were added for the same purpose).

Reply:
Yes, this is believed to be necessary. The checkout step will now ensure the submodules are fetched/present. Then, this step will update the submodule to point to the tag of the newly released version.

The failure that led to this change was in this step. When given this command to run:

```
git -C "ext/cli" checkout "v6.1.0"
```

The following error was produced:

```
error: pathspec 'v6.1.0' did not match any file(s) known to git
```

...which indicates that the submodule was not populated and pointing to the `cli` repo.

If the question is more about the inclusion of the `fetch-tags` option, then yes, that option may be superfluous. That is, it is clear that `fetch-tags` will ensure the tags for the `documentation` repo are fetched. It is not clear if the same setting applies to the tags of submodules. The decision to include it here was made to expedite the update of this repository and limit the number of iterations needed to come to a working solution. At worst, the option provides no additional value and yet also does not hurt anything. At best, it may solve the problem at hand.